mitiq/calibration/tests/test_settings.py

```diff
@@ -122,13 +122,73 @@ def test_basic_settings():
     assert ghz_problem.two_qubit_gate_count == 1
     assert ghz_problem.ideal_distribution == {"00": 0.5, "11": 0.5}
 
+    lines = str(ghz_problem).split("\n")
+    ghz_problem_dict = ghz_problem.to_dict()
+    for line in lines:
+        [title, value] = line.split(":", 1)
+        key = title.lower().replace(" ", "_")
+        value = value.strip()
+        assert key in ghz_problem_dict
+        assert value == str(ghz_problem_dict[key])
```

Comment:
Let's move this block (along with some of the changes below) into it's own test that checks the code related to string formatting. That way we can keep the tests small and modular.

Reply:
Done